src/build.js

```diff
@@ -0,0 +1,43 @@
+/* @flow */
+import path from 'path';
+import {createWriteStream} from 'fs';
+import streamToPromise from 'stream-to-promise';
+
+import * as fs from './util/promised-fs';
+import {zipDir} from './util/zip-dir';
+
+
+export default function build({sourceDir, buildDir}: Object) {
```

Comment:
I think it would be good if it didn't require additional syntax. ES6 is one thing, but it's a future standard. Adding non-standard syntax to me seems a step too far, just my 2 cents. It may also hinder contribution.

I wonder if there's any alternative way to get the benefits without compilation?

I suppose there's also an argument here having some integration testing even if the core tests in the tree are just unit tests for the reasons you describe. Might make the flow stuff a bit less necessary.